contracts/CollateralRegistry.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: bsl-1.1
+
+/*
+  Copyright 2020 Unit Protocol: Artem Zakharov (az@unit.xyz).
+*/
+pragma solidity ^0.7.1;
+pragma experimental ABIEncoderV2;
+
+
+import "./VaultParameters.sol";
+
+
+/**
+ * @title CollateralRegistry
+ **/
+contract CollateralRegistry is Auth {
+
+    event CollateralAdded(address indexed asset);
+    event CollateralRemoved(address indexed asset);
+
+    mapping(address => uint) public collateralId;
+
+    address[] _collaterals;
+    
+    constructor(address _vaultParameters) Auth(_vaultParameters) {}
+
+    function addCollateral(address asset) public onlyManager {
+        require(asset != address(0), "Unit Protocol: ZERO_ADDRESS");
+
+        if (_collaterals.length != 0) {
```

Comment:
Btw, here and removeCollateral could be simplified using `isCollateral`. Code reuse is good.